cogs/displayinfo.py

```diff
@@ -33,85 +33,113 @@ class DisplayInfo(commands.Cog):
+            after_string_id = str(mem_after.id)
+            if after_string_id not in self.timers:
+                self.timers[after_string_id] = {} # If the member is not already tracked, add them
+            if self.current_date not in self.timers[after_string_id]:
+                self.timers[after_string_id][self.current_date] = {} # If the current date is not present in the member's dictionary, add it
+            if mem_after.activity.name not in self.timers[after_string_id][self.current_date]: # If the member's current activity/game/status is not present, add it with initial values
+                self.timers[after_string_id][self.current_date][mem_after.activity.name] = {
                     "totalTime" : 0, 
                     "startTime" : 0,
-                    "updateTask" : None,
                 }
-            self.timers[mem_after.name][self.current_date][mem_after.activity.name]["startTime"] = datetime.now() # Record the start time of the member's current activity
+            self.timers[after_string_id][self.current_date][mem_after.activity.name]["startTime"] = datetime.now() # Record the start time of the member's current activity
             print("start:", self.timers)
-            self.timers[mem_after.name][self.current_date][mem_after.activity.name]["updateTask"] = asyncio.create_task(self.timed_update(mem_before, mem_after))
         elif (mem_before.activity and not mem_after.activity) or (mem_before.status != mem_after.status and mem_after.status == discord.Status.offline): # activity stopped or member
-            if mem_before.name in self.timers:
-                if self.current_date in self.timers[mem_before.name]:
-                    if mem_before.activity is not None and mem_before.activity.name in self.timers[mem_before.name][self.current_date]:
-                        if self.timers[mem_before.name][self.current_date][mem_before.activity.name]["updateTask"]:
-                            self.timers[mem_before.name][self.current_date][mem_before.activity.name]["updateTask"].cancel()
-                            try:
-                                await self.timers[mem_before.name][self.current_date][mem_before.activity.name]["updateTask"]
-                            except asyncio.CancelledError:
-                                print("task is now cancelled")
-                        self.timers[mem_before.name][self.current_date][mem_before.activity.name]["updateTask"] = None
-
-                        currentStartTime = self.timers[mem_before.name][self.current_date][mem_before.activity.name]["startTime"]
+            before_string_id = str(mem_before.id)
+            if before_string_id in self.timers:
+                if self.current_date in self.timers[before_string_id]:
+                    if mem_before.activity is not None and mem_before.activity.name in self.timers[before_string_id][self.current_date]:
```

Comment:
use `if mem_before.activity and mem_before.activity.name in self.....` instead of `is not None` check